src/hooks/internal/useResolvedLocaleOptions.ts

```diff
@@ -0,0 +1,67 @@
+import { Intl } from '@js-temporal/polyfill'
+import { useMemo } from 'react'
+import getValidLocale from '../../utils/getValidLocale'
+import { LocaleOptions } from '../useDatePicker'
+
+type UseResolvedLocaleOptionsHook = (
+    options: LocaleOptions
+) => LocaleOptions & { locale: string }
+
+/**
+ * A hook that returns the locale and locale options to be used by the calendar.
+ *
+ * The order of precedence of how the options are resolved should be:
+ *
+ * - Prop-values passed to the hook/component
+ * - Properties available from the system/user settings resource* (in the future)
+ * - Values coming from Intl.DateTimeFormat().resolvedOptions()
+ * - Some hardcoded defaults
+ *
+ * @param options
+ * @returns
+ */
+export const useResolvedLocaleOptions: UseResolvedLocaleOptionsHook = (
+    userOptions
+) => {
+    // validate that the passed locale is valid, use it if valid, otherwise, keep it undefined
+    const locale = getValidLocale(userOptions.locale)
+
+    const defaultDateTimeOptions = Intl?.DateTimeFormat?.().resolvedOptions?.()
+
+    const defaultUserOptions = useMemo(
+        () => ({
+            timeZone: defaultDateTimeOptions.timeZone || 'UTC',
+            numberingSystem: defaultDateTimeOptions.numberingSystem || 'latn',
+            locale: defaultDateTimeOptions.locale || 'en',
+            weekDayFormat: 'narrow' as const, // resolvedOptions seems to not return a value in all major browsers
+        }),
+        [defaultDateTimeOptions]
+    )
```

Comment:
I'm not sure how TypeScript is not throwing an error here....
```js
// With all this conditional chaining, in theory defaultDateTimeOptions could be undefined
const defaultDateTimeOptions = Intl?.DateTimeFormat?.().resolvedOptions?.()
// And if defaultDateTimeOptions is undefined, the line below would throw an error
// because it would be reading a property of undefined
timeZone: defaultDateTimeOptions.timeZone || 'UTC',
```

Probably all the references to `defaultDateTimeOptions` properties should be conditionally chained too.... `defaultDateTimeOptions?.propertyName`

Reply:
Also, in general, it would be good to change all the `||` to operators to the `??` operator which is more apt when defining fallback values.

Reply:
Finally, for this line:
```js
weekDayFormat: 'narrow' as const, // resolvedOptions seems to not return a value in all major browsers
```

I suggest changing it to:
```js
weekDayFormat: defaultDateTimeOptions?.weekDayFormat || 'narrow',
```

This is more consistent, and might prove useful for the future when browser decide to return a value there.